src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -256,6 +256,16 @@ def _validate_vm_create_nics(namespace):
 
 def _validate_vm_create_auth(namespace):
 
+    if len(namespace.admin_username) < 6 or namespace.admin_username.lower() == 'root':
+        # prompt for admin username if inadequate
+        from azure.cli.core.prompting import prompt, NoTTYException
+        try:
+            logger.warning("Cannot use provided admin username: %s. Admin username should be at "
```

Comment:
Would consider removing 'provided' from the message.
If I run `az vm create -g myrg -n myname --image ubuntults`, I never did provide an admin username as it was implicit.